editor/src/components/window/file.rs

```diff
@@ -182,22 +182,10 @@ impl FileImplementedEditor for super::EchidnaWindow {
     }
 
     fn action_save_file(&self) -> Result<(), Box<dyn Error>> {
-        match self.get_current_tab() {
-            Ok(page) => {
-                let page: EchidnaCoreEditor = page;
-                match page.file() {
-                    Some(file) => {
-                        let location = file.location();
-                        match page.save_file(None) {
-                            Ok(_) => Ok(()),
-                            Err(e) => Err(e),
-                        }
-                    }
-
-                    None => self.action_save_file_as(),
-                }
-            }
-            Err(e) => Err(e),
+        let page: EchidnaCoreEditor = self.get_current_tab()?;
+        match page.file() {
+            Some(_) => page.save_file(file.location()),
```

Comment:
I think we need to change the `Some(_)` to `Some(file)`
```
error[E0423]: expected value, found macro `file`
   --> editor/src/components/window/file.rs:187:39
    |
187 |             Some(_) => page.save_file(file.location()),
    |                                       ^^^^ not a value
```

Reply:
done https://github.com/null8626/Echidna/commit/a094a21e4c8a3771221de02c994b773c5e6edc56